homeassistant/components/myio/sensor.py

```diff
@@ -0,0 +1,112 @@
+import logging
+
+from homeassistant.const import CONF_NAME, TEMP_CELSIUS
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import slugify
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(seconds=3)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add a sensor entity from a config_entry."""
+
+    _server_name = slugify(config_entry.data[CONF_NAME])
+    _server_status = hass.states.get(_server_name + ".state").state
+    _server_data = hass.data[_server_name]
+    _sensors = _server_data["sensors"]
+
+    sensor_entities = []
+    if _server_status.startswith("Online"):
+        try:
+            for _sensor in _sensors:
+                _actual_sensor = _sensors[_sensor]
+                _sensor_id = _actual_sensor["id"]
+                if (
+                    0 < _sensor_id <= 100
+                    or (100 < _sensor_id < 200 and _actual_sensor["hum"] != 0)
+                    or (_sensor_id == 200)
+                ):
+                    sensor_entities.append(
+                        MyIOSensor(hass, _sensor_id, _server_data, _server_name)
+                    )
+        except (ValueError, Exception):  # pylint: disable=broad-except
+            _LOGGER.debug("%s - No sensor data", _server_name)
+
+        async_add_entities(sensor_entities)
+
+
+class MyIOSensor(Entity):
+    """Implementation of a myIO-Server."""
+
+    def __init__(self, hass, _sensor_id, _server_data, _server_name):
+        """Initialize the sensor."""
+        self._sensor_id = _sensor_id
+        self._server_name = _server_name
+        self._server_data = _server_data
+        self._sensors = _server_data["sensors"]
+        self._number = 0
+        self._state = 0
+
+        # initialize the correct number in the sensors list of the sensor
+        for _sensor in self._sensors:
+            if self._sensors[_sensor]["id"] == _sensor_id:
+                self._number = _sensor
+                break
+        self.entity_id = f"sensor.{_server_name}_{self._sensor_id}"
+        self._name = self._sensors[str(self._number)]["description"]
+        # initialize if the sensor is thermo-, humidity-, or imp counter"""
+        if int(self._number) < 100:
+            self._state = self._sensors[str(self._number)]["temp"] / 100
+        elif int(self._number) < 200:
+            self._state = self._sensors[str(self._number)]["hum"] / 10
+        else:
+            self._state = self._sensors[str(self._number)]["imp"]
+
+    @property
+    def scan_interval(self):
+        """Return the unique id."""
+        return SCAN_INTERVAL
+
+    @property
+    def unique_id(self):
+        """Return unique ID for light."""
+        return f"server name = {self._server_name}, id = {self._sensor_id}"
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        if int(self._number) >= 200:
+            return "Imp.Counter"
+        return self._name
```

Comment:
Determining the name is now spread in this method and in the class init. Maybe consolidate that logic so the code becomes more clear?